libsolidity/analysis/OverflowBounds.h

```diff
@@ -0,0 +1,59 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/**
+ * @author Rhett <roadriverrail@gmail.com>
+ * @date 2017
+ * Overflow range container.
+ */
+
+#pragma once
+
+#include <libsolidity/ast/Types.h>
+#include <libsolidity/interface/Utils.h>
+#include <libsolidity/parsing/Token.h>
+#include <libdevcore/Common.h>
+#include <limits>
+
+namespace dev
+{
+namespace solidity
+{
+
+class OverflowBounds
```

Comment:
No, probably not.  I had wanted to make a generic class that would be useful for both reasoning about integers and booleans.  The more I work with it, however, the worse that idea even seems.  I am starting to feel like I should just get rid of the parent class until I know what I'm doing with it anyway.  What do you think?